misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -3513,8 +3513,17 @@ private <T extends Node> T formatNode(T node, int trailingWS, int trailingNL) {
             env.trailingNL = trailingNL;
             env.trailingWS = trailingWS;
 
+            // Cache the current node and parent before format.
+            // Because reference to the nodes will change after modifying.
+            T oldNode = node;
+            Node parent = node.parent();
+
             node = (T) node.apply(this);
 
+            if (shouldWrapLine(oldNode, parent) && options.getLineWrapping()) {
```

Comment:
IMO we can change the precedence of the checks for better performance (having the `options.getLineWrapping()` check first should prevent executing the relatively expensive operation, i.e. `shouldWrapLine()` every-time). WDYT?

Reply:
Ack, made the suggested change.